### DISCUSSION TITLE
How to know if the current screen is within a presentation modal?

### DISCUSSION THREAD

**Discussion by KishoreBarik** [react-navigation/react-navigation#10420] [Q&A]
I have a use case where the screen component is being used in normal presentation as well as within a modal presentation flow(Parent to Parent stack is modal). How can I determine (using the library help) if the screen is within a modal or not.
I see `ModalPresentationContext` is there in Stack navigator, but this is not exported. Also there are some other functions like `getIsModal` which makes use of `scene.descriptor.options` to determine the same. But neither of them is exposed.

**Comment by a-eid** [comment]
sounds like a very good question, especially with the native stack.

**Comment by adamhari** [comment]
Also curious, this is useful for managing the status bar and safe area differently on screens which are sometimes a modal.

**Comment by cwooldridge1** [comment]
With react-navigation there is a type field that you can access that will tell you if you are looking at a tab, stack etc...
```
import { useNavigation } from "@react-navigation/native";

const navigation = useNavigation();
navigation.getState()?.type; // tab, stack etc..
```
